polars/polars-core/src/chunked_array/strings.rs

```diff
@@ -2,6 +2,36 @@ use crate::chunked_array::kernels::strings::string_lengths;
 use crate::prelude::*;
 use arrow::compute::kernels::substring::substring;
 use regex::Regex;
+#[cfg(feature = "extract_jsonpath")]
+use {
+    jsonpath_lib::Compiled,
+    serde_json::{from_str, Value},
+    std::borrow::Cow,
+    std::string::String,
+};
+
+#[cfg(feature = "extract_jsonpath")]
+fn extract_json<'a>(expr: &'a Compiled, json_str: Option<&'a str>) -> Option<Cow<'a, str>> {
+    if let Some(value) = json_str {
```

Comment:
`if let Some(..) else { None }`. I think we can replace this with an `Option::map`

Reply:
Right, I'll use `and_then`, learning new things by the day :D